src/biokbase/narrative/jobs/job.py

```diff
@@ -199,7 +226,7 @@ def __getattr__(self, name):
                 # and need the state refresh.
                 # But KBParallel/KB Batch App jobs may not have the
                 # batch_job field
-                self.state(force_refresh=True).get(
+                self.refresh_state(force_refresh=True).get(
```

Comment:
I am wondering if it would be better to get rid of the "magic" in these attributes, so that they just return the current value, rather than automatically updating. Explicitly requesting that jobs be updated fits better with "the principle of least surprise", and it would also make the code simpler because you don't have to worry about getting into some loop of endless updates. The only thing is that you'd need to take a look at where these various attributes have been called and ensure that there's an appropriate job refresh called.

Reply:
Sensible

Reply:
Though I want to keep this a tiny PR!

Reply:
Can you take a look and see how much it would take to remove the "magic" attributes? There are only a few that do trigger updates, and we can also fix the lack of a `status` attribute at the same time. It should be easy enough just to make a separate PR with this content.

Reply:
Yeah I think there's too much magic ...